File: euler10.py
def SumOfPrimes(n):
    prime=[]
    for i in range(n+1):
        prime.append(1)
    s=0
    p=2
    while(p*p<=n):
        if (prime[p] == 1):
                i=p*p
                while(i<=n):
                    prime[i]=0
                    i+=p
        p+=1
    for i in range(2,n):
        if(prime[i]==1):
            s=s+i;
    print(s)

n=int(input("Enter limit"))
SumOfPrimes(n);

File: euler10.cpp
// C++ program to print all primes smaller than or equal to n using Sieve of Eratosthenes
#include <iostream>
#include<string.h>//for memset function to set memory to a particular value
using namespace std;

void SumOfPrimes(int n)
{
    bool prime[n+1];
    memset(prime, true, sizeof(prime));
    int s=0;
    for (int p=2; p*p<=n; p++)
    {
        if (prime[p] == true)
        {
            for (int i=p*p; i<=n; i += p)
                prime[i] = false;
        }
    }
    for(int i=2;i<=n;i++)
    {
          if(prime[i])
          s=s+i;
    }
    cout<<s<<endl;
}

// Driver Program to test above function
int main()
{
    int n;
    cout<<"Enter limit: ";
    cin>>n;
    SumOfPrimes(n);
    return 0;
}
